### DISCUSSION TITLE
Integration testing with Aspire

### DISCUSSION THREAD

**Discussion by Dazfl** [microsoft/aspire#6734] [Q&A]
I'm looking for some guidance around integration testing with an application built with Aspire.

In projects I have previously worked on, my team and I have set up a test project (xUnit) where we spin up a MS SQL server instance in docker, but create a database per feature (or collection of tests).  We do this via docker compose, and we use Reqnroll (formerly Specflow) to split our tests into features, etc.  This helps keep the tests isolated from each other, making it a lot easier to debug.

I'm wanting to know if this concept is possible with Aspire.  I love how the Apphost can spin up all the infrastructure needed for an application, but I having trouble working out if I can spin up a single instance of the database server, but create a database per collection of tests.

Are there any examples of where this has been done with Aspire?

**Comment by epgeroy** [comment]
@Dazfl, you might want to check this: https://learn.microsoft.com/en-us/dotnet/aspire/testing/write-your-first-test?pivots=xunit
